FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -0,0 +1,291 @@
+// Copyright 2019 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "FIRAppDistribution.h"
+#import "FIRAppDistribution+Private.h"
+
+#import <FirebaseCore/FIRAppInternal.h>
+#import <FirebaseCore/FIRComponent.h>
+#import <FirebaseCore/FIRComponentContainer.h>
+#import <FirebaseCore/FIROptions.h>
+
+#import <AppAuth/AppAuth.h>
+#import <FIRAppDistributionAppDelegateInterceptor.h>
+#import <GoogleUtilities/GULAppDelegateSwizzler.h>
+#import <UIKit/UIKit.h>
+
+/// Empty protocol to register with FirebaseCore's component system.
+@protocol FIRAppDistributionInstanceProvider <NSObject>
+@end
+
+@interface FIRAppDistribution () <FIRLibrary, FIRAppDistributionInstanceProvider>
+@end
+
+@implementation FIRAppDistributionRelease
+- (instancetype)init {
+  self = [super init];
+
+  return self;
+}
+@end
+
+@implementation FIRAppDistribution
+
+// The OAuth scope needed to authorize the App Distribution Tester API
+NSString *const OIDScopeTesterAPI = @"https://www.googleapis.com/auth/cloud-platform";
+
+// The App Distribution Tester API endpoint used to retrieve releases
+NSString *const ReleasesEndpointURL =
+    @"https://firebaseapptesters.googleapis.com/v1alpha/devices/-/testerApps/%@/releases";
+
+@synthesize isTesterSignedIn = _isTesterSignedIn;
+
+#pragma mark - Singleton Support
+
+- (instancetype)initWithApp:(FIRApp *)app appInfo:(NSDictionary *)appInfo {
+  self = [super init];
+
+  if (self) {
+    self.safariHostingViewController = [[UIViewController alloc] init];
+
+    // Save any properties here
+    NSLog(@"APP DISTRIBUTION STARTED UP!");
+
+    [GULAppDelegateSwizzler proxyOriginalDelegate];
+
+    FIRAppDistributionAppDelegatorInterceptor *interceptor =
+        [FIRAppDistributionAppDelegatorInterceptor sharedInstance];
+    [GULAppDelegateSwizzler registerAppDelegateInterceptor:interceptor];
+  }
+
+  // TODO: Lookup keychain to load auth state on init
+  _isTesterSignedIn = self.authState ? YES : NO;
+  return self;
+}
+
++ (void)load {
+  [FIRApp registerInternalLibrary:(Class<FIRLibrary>)self
+                         withName:@"firebase-appdistribution"
+                      withVersion:@"0.0.0"];  // TODO: Get version from podspec
```

Comment:
Just FYI, when you get to this.

You can set a variable with the version information from your podspec via xcconfig:
https://github.com/firebase/firebase-ios-sdk/blob/e338339c594a80a44dbc2013ff188bb3ce225580/FirebaseCrashlytics.podspec#L58

Reply:
Done!